### PR TITLE
Add DO one-click app introduction post

### PR DESCRIPTION
**Pull request by rgee0**
Signed-off-by: Richard Gee <richard@technologee.co.uk>

### CODE REVIEW

File: _posts/2019-02-02-digitalocean-one-click.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
+---
+title: "Announcing DigitalOcean's OpenFaaS One-click"
+description: Earlier this week saw the release of the first OpenFaaS One-click application on DigitalOcean.  Richard Gee guides you through just how simple it is to create your first OpenFaaS function on DigitalOcean.
```
Comment (alexellis) [inline _posts/2019-02-02-digitalocean-one-click.md]: The description will be part of our SEO for the post so let's drop "Earlier this week saw the release of the first "

Comment (alexellis) [inline _posts/2019-02-02-digitalocean-one-click.md]: Join Richard Gee as he guides you through / walks you through the experience?

File: _posts/2019-02-02-digitalocean-one-click.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+date: 2019-02-02
+image: /images/digitalocean-one-click/balance-macro-ocean.jpg
+categories:
```
Comment (alexellis) [inline _posts/2019-02-02-digitalocean-one-click.md]: + `- tutorial`

File: _posts/2019-02-02-digitalocean-one-click.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+  - developer-experience
+  - swarm
+  - python
+author_staff_member: richard
+dark_background: false
+---
+
+I've written previously about how you can be [up and running in 5 minutes](/blog/deploy-digitalocean-ansible.md) with OpenFaaS on DigitalOcean.  With the news this week that DigitalOcean are now offering an OpenFaaS One-click application this time shortens further and increases the number of cost effective ways in which to run OpenFaaS on DigitalOcean:
```
Comment (alexellis) [inline _posts/2019-02-02-digitalocean-one-click.md]: In <month> <year> I a blog post on how you could set up OpenFaaS on DigitalOcean with Docker Swarm or Kubernetes in 5 minutes.

Comment (alexellis) [inline _posts/2019-02-02-digitalocean-one-click.md]: Explain that we collaborated with DigitalOcean to provide the one-click image. (this makes it more deliberate and shows it was a partnership)

File: _posts/2019-02-02-digitalocean-one-click.md [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+
+* [Vanilla Droplets running Swarm or Kubernetes](/blog/deploy-digitalocean-ansible.md)
```
Comment (alexellis) [inline _posts/2019-02-02-digitalocean-one-click.md]: s/Vanilla/Ubuntu Linux

File: _posts/2019-02-02-digitalocean-one-click.md [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+* Using the [DigitalOcean Kubernetes Service (DOKS)](https://blog.alexellis.io/digitalocean-kubernetes-engine/)
+* One-click application creating a single Ubuntu node Swarm-based instance
+
+In this post we'll guide you through creating your first OpenFaaS function using the One-Click application.
```
Comment (alexellis) [inline _posts/2019-02-02-digitalocean-one-click.md]: What do they call it in their UI, is it "One-Click" or "Droplet Image" or etc?

File: _posts/2019-02-02-digitalocean-one-click.md [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+
+### Pre-Requisites
+
+* A DigitalOcean account
+
+An obvious pre-req is a [DigitalOcean](https://www.digitalocean.com/) account.  A DigitalOcean account with $100/60 days of credit can be  [created here](https://m.do.co/c/2962aa9e56a1).  This provides a generous amount of credit with the most economic Droplet size being just $5 a month.
```
Comment (alexellis) [inline _posts/2019-02-02-digitalocean-one-click.md]: Which referral link is this?

File: _posts/2019-02-02-digitalocean-one-click.md [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+
+* The OpenFaaS CLI
+
+The OpenFaaS CLI (or `faas-cli`) is now a core piece of the OpenFaaS developer-experience and our users run it locally and in their CI jobs.  We'll use it to create and deploy our function onto the One-click Droplet.  The OpenFaaS docs provide all the detail you need to [get the CLI](https://docs.openfaas.com/cli/install/).
```
Comment (alexellis) [inline _posts/2019-02-02-digitalocean-one-click.md]: Sometimes I see "One-Click" in other places I see "One-click"

File: _posts/2019-02-02-digitalocean-one-click.md [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+
+### Create the Droplet
```
Comment (alexellis) [inline _posts/2019-02-02-digitalocean-one-click.md]: s/the/your

File: _posts/2019-02-02-digitalocean-one-click.md [hunk 9]
```diff
@@ -0,0 +1,193 @@
...
+
+[Log in](https://cloud.digitalocean.com/login) to your DigitalOcean account and head over to the create button, which details the products currently being offered by DigitalOcean.  The One-click apps reside under Droplets.
+
+![Create button menu](/images/digitalocean-one-click/create-button.png)
```
Comment (alexellis) [inline _posts/2019-02-02-digitalocean-one-click.md]: Good screenshots. What size are these roughly?

File: _posts/2019-02-02-digitalocean-one-click.md [hunk 10]
```diff
@@ -0,0 +1,193 @@
...
+
+On the resulting menu select the One-click apps option which will present a number of One-click apps.
+
+![Droplet menu](/images/digitalocean-one-click/droplet-menu.png)
+
+We'll choose the one labelled OpenFaaS on 18.04.  This will create a Swarm-based single node OpenFaaS instance on Ubuntu 18.04.
+
+![OpenFaaS One-click](/images/digitalocean-one-click/openfaas-oneclick.jpg)
+
+From here choose a Droplet size and region.  I'd recommend `4 GB / 2 CPUs / 80 GB SSD` and I tend to pick the region closest to me - being in the UK this is London.  Finally, choose your SSH key - if you need to generate and add a new key then please follow the [instructions on the previous blog](/blog/deploy-digitalocean-ansible/#create-and-upload-an-ssh-key).  With your chosen configuration click Create.
```
Comment (alexellis) [inline _posts/2019-02-02-digitalocean-one-click.md]: I would go with something like:

> You can run OpenFaaS on the cheapest node available for just 5 USD per month, but for the best experience and some future proofing I would recommend using the 4GB / etc.

File: _posts/2019-02-02-digitalocean-one-click.md [hunk 11]
```diff
@@ -0,0 +1,193 @@
...
+
+> Note: ensure you have made the appropriate private key available to the client you are using.
+
+Once logged in follow the instructions; at the time of writing this is simply to press Enter.
+
+![Logged in](/images/digitalocean-one-click/terminal.png)
+
+A number of configuration actions will be performed for you, and the result will be a running Swarm-based OpenFaaS instance.  Make a note of the credentials - in particular the final line which will set our credentials in the `faas-cli` - we'll need this when we deploy our function.  We can now close the connection to the Droplet.
+
+* On your local machine set the `OPENFAAS_URL`:
```
Comment (alexellis) [inline _posts/2019-02-02-digitalocean-one-click.md]: We could mention they can edit their `.bashrc` file or `.bash_profile` to retain the setting / make permanent.

File: _posts/2019-02-02-digitalocean-one-click.md [hunk 12]
```diff
@@ -0,0 +1,193 @@
...
+| |_| | |_) |  __/ | | |  _| (_| | (_| |___) |
+ \___/| .__/ \___|_| |_|_|  \__,_|\__,_|____/
+      |_|
+
+
+Function created in folder: hello-digitalocean
+Stack file written: hello-digitalocean.yml
+```
+
+* Rename `hello-digitalocean.yml` to `stack.yml`:
```
Comment (alexellis) [inline _posts/2019-02-02-digitalocean-one-click.md]: Not about the blog, but should we change the CLI to look first for stack.yml then any YAML file found in the directory to stop people having to rename to stack.yml in every blog post / guide?

File: _posts/2019-02-02-digitalocean-one-click.md [hunk 13]
```diff
@@ -0,0 +1,193 @@
...
+
+```
+$ mv hello-digitalocean.yml stack.yml
+```
+
+* Edit `hello-digitalocean/handler.py` to add our function's code:
+
+```
```
Comment (alexellis) [inline _posts/2019-02-02-digitalocean-one-click.md]:
```python

File: _posts/2019-02-02-digitalocean-one-click.md [hunk 14]
```diff
@@ -0,0 +1,193 @@
...
+    """handle a request to the function
+    Args:
+        req (str): request body
+    """
+    return 'Hello from OpenFaaS on DigitalOcean!'
+```
+
+* Save the handler and run `faas-cli up` to build, push and deploy:
+
+```
```
Comment (alexellis) [inline _posts/2019-02-02-digitalocean-one-click.md]:
```sh
```sh

File: _posts/2019-02-02-digitalocean-one-click.md [hunk 15]
```diff
@@ -0,0 +1,193 @@
...
+
+This is great as a starting point to quickly getting hands on with OpenFaaS.  With an instance up and running you are ideally placed to work through the [OpenFaaS Workshop](https://github.com/openfaas/workshop).  If you've followed this article fully then you should be able to skip straight on to [Lab 2](https://github.com/openfaas/workshop/blob/master/lab2.md) - remember to substitute any workshop references to `127.0.0.1` with the IP address of your Droplet.
+
+Taking it further we could look at how to configure a domain name to point at the instance and enable transit encryption through use of a service like [Let's Encrypt](https://letsencrypt.org/).
+
+
+* For questions, comments and suggestions follow us on [Twitter @openfaas](https://twitter.com/openfaas) and join the [Slack community](https://docs.openfaas.com/community).
+
+* Learn how to contribute on YouTube: [How To Contribute to OpenFaaS](https://www.youtube.com/watch?v=kOgHjU38Efg)
```
Comment (alexellis) [inline _posts/2019-02-02-digitalocean-one-click.md]: Can we do a video embed? Does that work on here? @johnmccabe ?

File: _posts/2019-02-02-digitalocean-one-click.md [hunk 16]
```diff
@@ -0,0 +1,193 @@
+---
+title: "Announcing DigitalOcean's OpenFaaS One-click"
+description: Earlier this week saw the release of the first OpenFaaS One-click application on DigitalOcean.  Richard Gee guides you through just how simple it is to create your first OpenFaaS function on DigitalOcean.
+date: 2019-02-02
+image: /images/digitalocean-one-click/balance-macro-ocean.jpg
```
Comment (alexellis) [inline _posts/2019-02-02-digitalocean-one-click.md]: Please see my guidelines for resizing the lead image to ~= 280KB in the README.md

Comment (alexellis) [reply _posts/2019-02-02-digitalocean-one-click.md]: I like this theme. Can we find an image that has a sea-blue with some waves to show the ocean theme? Or something a bit more vibrant?

https://www.pexels.com/photo/photography-of-barrel-wave-1298684/
https://www.pexels.com/photo/ocean-wave-1656579
https://www.pexels.com/photo/beach-calm-clouds-horizon-459522/
https://www.pexels.com/photo/beach-beautiful-blue-sky-blue-water-459252/

File: _posts/2019-02-02-digitalocean-one-click.md (hunk 5 above)
Comment (rgee0) [reply _posts/2019-02-02-digitalocean-one-click.md]: One-click

File: _posts/2019-02-02-digitalocean-one-click.md (hunk 6 above)
Comment (rgee0) [reply _posts/2019-02-02-digitalocean-one-click.md]: It should be the same one as the Ansible post.

File: _posts/2019-02-02-digitalocean-one-click.md (hunk 12 above)
Comment (rgee0) [reply _posts/2019-02-02-digitalocean-one-click.md]: Depends how people lay out their functions.  If they have multiple functions / multiple ymls in the same top level folder then I'm not sure what this behaviour would look like.

File: _posts/2019-02-02-digitalocean-one-click.md (hunk 16 above)
Comment (rgee0) [reply _posts/2019-02-02-digitalocean-one-click.md]: I was trying to be more subtle than a picture of an ocean.  Quite liked the idea of a stack in the foreground.

**Comment by rgee0** [comment]
I've also updated the credit text on the original post since this post links back.

File: _posts/2019-02-02-digitalocean-one-click.md (hunk 15 above)
Comment (alexellis) [reply _posts/2019-02-02-digitalocean-one-click.md]: Can you do the embedding?

File: _posts/2019-02-02-digitalocean-one-click.md [hunk 17]
```diff
@@ -0,0 +1,197 @@
...
+```sh
+$ mkdir -p ~/openfaas/functions/ && cd ~/openfaas/functions/
+```
+
+* Create a function called `hello-digitalocean` using the CLI:
+
+> Replace the `rgee0` prefix with your own Docker hub account name so as to enable the ensuing `push` action.
+
+```sh
+$ faas-cli new --lang=python --prefix=rgee0 hello-digitalocean
```
Comment (alexellis) [inline _posts/2019-02-02-digitalocean-one-click.md]: Can you do `python3` given the deprecation notice?

**Comment by alexellis** [comment]
Richard please can you resize these images to 50%? They look huge and out of proportion on my local environment (Safari)

Alex

screenshot 2019-02-04 at 12 58 43

**Pull request merged**